### PR TITLE
Add implementation for room-invite event

### PR DESCRIPTION
**Pull request by windmemory**
Currently this is a prototype, not finished version.

### CODE REVIEW

**Comment by windmemory** [comment]
I was using `axios` to do a post with the url returned back from `WXGetRequestToken` call. But it triggered websocket close event, then throw out an exception. The exception as below:
```shell
TypeError: Cannot read property 'request' of undefined
    at RedirectableRequest._performRequest (/Users/yuangao/git/wechaty/wechaty-puppet-padchat/node_modules/follow-redirects/index.js:132:24)
    at RedirectableRequest._processResponse (/Users/yuangao/git/wechaty/wechaty-puppet-padchat/node_modules/follow-redirects/index.js:222:10)
    at ClientRequest.RedirectableRequest._onNativeResponse (/Users/yuangao/git/wechaty/wechaty-puppet-padchat/node_modules/follow-redirects/index.js:39:10)
    at Object.onceWrapper (events.js:273:13)
    at ClientRequest.emit (events.js:182:13)
    at ClientRequest.EventEmitter.emit (domain.js:442:20)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:546:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:109:17)
    at TLSSocket.socketOnData (_http_client.js:432:20)
    at TLSSocket.emit (events.js:182:13)
    at TLSSocket.EventEmitter.emit (domain.js:442:20)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at TLSSocket.Readable.push (_stream_readable.js:219:10)
    at TLSWrap.onread (net.js:635:20)
```

Seems like error with `axios`, but I can not find anyone talked about this in `axios` repo. Then I decide to try with `request-promise`, but after I installed `request-promise`, all the npm link I created before seems got reset, not sure why, but setup npm link is time consuming, so I decide to have the change in `wechaty` and `wechaty-puppet` get published, then work on the impl. 

Anyway, here is the workable prototype you asked, please let me know what do you think.

**Comment by zixia** [comment]
Could you please deploy your workable prototype on a c9 IDE, so that we can run it? Because this PR can not **workable** without a runnable environment.

If you have any trouble with your c9 account, we have a public c9 ide at <https://ide.c9.io/zixia/wechaty-bug-reproducer> and you can apply a write permission for that. But please be careful that it's public so everyone can read it.

**Comment by zixia** [comment]
Yes, I agree with you.

Amazon sucks.

**Comment by zixia** [comment]
Is there any documentation, or at least a note of how to run your code?

**Comment by windmemory** [comment]
@zixia Please let me know what do you think about this approach :)

**Comment by windmemory** [comment]
@zixia Do a `ts-node bot.ts` in the root folder.

**Comment by windmemory** [comment]
English version will also return back in Chinese, I've tested that.

**Comment by windmemory** [comment]
@zixia  ping

**Comment by zixia** [comment]
Will look into it for you tonight.

**Comment by windmemory** [comment]
Hi @zixia did you look into it?

**Comment by zixia** [comment]
Please see my reply at <https://github.com/Chatie/wechaty/issues/1492#issuecomment-406840380>

File: package.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
     "tslint": "^5.10.0",
     "tslint-config-standard": "^7.1.0",
     "typescript": "^2.9.2",
-    "wechaty": "^0.17.133"
+    "wechaty": "^0.19.107"
```
Comment (zixia) [inline package.json]: I have to notify you that this is a BREAKING change and you should bump your MINOR version number when you **changing the dependence to wechaty 0.17 to 0.19**

**Comment by windmemory** [comment]
Bumped

**Pull request merged**